integration/apps/pennant/0003-Added-GEOPM-region-markup.patch

```diff
@@ -0,0 +1,404 @@
+From 31d5ec851755fb6a948b41bae9a3bacb3fd6f2ab Mon Sep 17 00:00:00 2001
+From: Fuat Keceli <fuat.keceli@intel.com>
+Date: Tue, 1 Dec 2020 17:44:13 -0800
+Subject: [PATCH 3/3] Added GEOPM region markup
+
+#  Copyright (c) 2015, 2016, 2017, 2018, 2019, 2020, Intel Corporation
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions
+#  are met:
+#
+#      * Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in
+#        the documentation and/or other materials provided with the
+#        distribution.
+#
+#      * Neither the name of Intel Corporation nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+Signed-off-by: Fuat Keceli <fuat.keceli@intel.com>
+---
+ Makefile            |  6 ++++
+ src/Driver.cc       | 28 +++++++++++++++++-
+ src/Hydro.cc        | 85 ++++++++++++++++++++++++++++++++++++++++++++++++++++-
+ src/PennantGeopm.cc | 56 +++++++++++++++++++++++++++++++++++
+ src/PennantGeopm.hh | 33 +++++++++++++++++++++
+ src/main.cc         |  3 ++
+ 6 files changed, 209 insertions(+), 2 deletions(-)
+ create mode 100644 src/PennantGeopm.cc
+ create mode 100644 src/PennantGeopm.hh
+
+diff --git a/Makefile b/Makefile
+index a0611ef..369d119 100644
+--- a/Makefile
++++ b/Makefile
+@@ -40,6 +40,12 @@ CXXFLAGS := $(CXXFLAGS_OPT)
+ ifdef USEGEOPM
+ 	LDFLAGS += $(GEOPM_LDFLAGS) $(GEOPM_LDLIBS)
+ 	CXXFLAGS += $(GEOPM_CFLAGS) -DUSEGEOPM -DEPOCH_TO_OUTERLOOP=$(EPOCH_TO_OUTERLOOP)
++ifdef USEGEOPMMARKUP
++	CXXFLAGS += -DUSEGEOPMMARKUP
++endif
++ifdef USEGEOPMMARKUP_2
```

Comment:
do we need both sets of markup? Is one of these going to become the canonical version?

Reply:
I removed the 2nd set of markup.